scipy/signal/_signaltools.py

```diff
@@ -3910,19 +3949,23 @@ def vectorstrength(events, period):
     # we need to know later if period was originally a scalar
     scalarperiod = not period.ndim
 
-    events = np.atleast_2d(events)
-    period = np.atleast_2d(period)
-    if (period <= 0).any():
+    events = xpx.atleast_nd(events, ndim=2, xp=xp)
+    period = xpx.atleast_nd(period, ndim=2, xp=xp)
+    if xp.any(period <= 0):
         raise ValueError('periods must be positive')
 
     # this converts the times to vectors
-    vectors = np.exp(np.dot(2j*np.pi/period.T, events))
+    I2pi = xp.asarray(2j*xp.pi, dtype=xp.complex64
+                        if period.dtype == xp.float32
+                        else xp.complex128)
```

Comment:
following https://github.com/data-apis/array-api/pull/848 this could look like

```python
I2pi = xp.asarray(2j*xp.pi, dtype=xp.astype(period, 'complex floating').dtype)
```

https://github.com/data-apis/array-api-extra/issues/13 was also suggesting something like

```python
I2pi = xp.asarray(2j*xp.pi, dtype=xpx.complex_dtype(period.dtype))
```

Reply:
Oh for good green Earth, I seriously hope for `xp.exp(1j*xp.linspace(0, pi, 3)/pi` just working.